scripts/promote-catalog.py

```diff
@@ -135,18 +135,33 @@ def update_makefile(
     """
     to_commit = ""
 
-    relative_makefile_path = os.path.join("operators", operator_name, "Makefile")
-    makefile_path = os.path.join(repo_dir, relative_makefile_path)
+    relative_makefile_path = Path("operators", operator_name, "Makefile")
+    makefile_path = Path(repo_dir, relative_makefile_path)
     with open(makefile_path, "r") as makefile:
         makefile_content = makefile.readlines()
 
     makefile_out = []
     for line in makefile_content:
-        if line.startswith("OCP_VERSIONS"):
-            if target_version not in line:
-                # assuming line looks like:
-                # `OCP_VERSIONS=$(shell echo "v4.12 v4.13 v4.14 v4.15" )`
-                split_point = line.rindex('"')
+        # search for variable assignment
+        if line.startswith("OCP_VERSIONS="):
+            # find the quoted part of the line
+            # assuming line might looks like:
+            # `OCP_VERSIONS=$(shell echo "v4.12 v4.13 v4.14 v4.15" )`
+            # OR
+            # `OCP_VERSIONS="v4.12 v4.13 v4.14 v4.15"`
```

Comment:
`OCP_VERSIONS=$(shell echo "v4.12 v4.13 v4.14 v4.15" )` and `OCP_VERSIONS="v4.12 v4.13 v4.14 v4.15"` are not the same in a Makefile.
The equivalent version would be `OCP_VERSIONS=v4.12 v4.13 v4.14 v4.15` (without the quotes).
Also there can be any number of spaces around the `=`.

Reply:
I changed the logic to use regex to match the start of the line and then capture the versions with updated regex to make it independent on quotes. Thanks for explaining the Makefile specifics.